aten/src/ATen/cuda/CUDAEvent.h

```diff
@@ -62,8 +78,27 @@ struct AT_CUDA_API CUDAEvent {
   cudaEvent_t event() const { return event_; }
 
   // Note: cudaEventQuery can be safely called from any device
+  bool query() const {
+    if (!is_created_) {
+      return true;
+    }
+
+    cudaError_t err = cudaEventQuery(event_);
+    if (err == cudaSuccess) {
+      return true;
+    } else if (err != cudaErrorNotReady) {
+      C10_CUDA_CHECK(err);
+    }
+
+    return false;
+  }
+
   bool happened() const {
-    return (was_recorded_ && cudaEventQuery(event_) == cudaSuccess);
+    if (was_recorded_) {
```

Comment:
I think you should remove `happened()` since it is the same as `query()`, but is incorrect if the Event was opened from an IPC handle.